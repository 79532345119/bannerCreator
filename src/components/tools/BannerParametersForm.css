.tools_elem {
    display: block;
    border: blue 1px solid;
    border-radius: 5px;
    padding: 1rem;
    margin: 0 auto 10px;
}
.tools_elem:hover {
    border: blue 2px solid;
}
.tools_elem input {
    display: block;
    width: 100%;
    margin-bottom: 5px;
    border: inherit;
}
.tools_elem button {
    display: block;
    width: 100%;
    padding: 5px;
    margin: 10px 0;
}
.tools_elem span {
    display: flex;
    justify-content: space-between;
    align-items: center;
}
.tools_elem span input.sm {
    width: 20%;
    margin-left: 10px;
}
.tools_elem span button {
    width: auto;
}
.tools_elem span label {
    width: 100%;
}

