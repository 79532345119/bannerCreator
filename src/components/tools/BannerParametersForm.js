import React, { useState } from "react"
import "./BannerParametersForm.css" 


export const BannerParametersForm = (props) => {
    
    const [width, setWidthValue] = useState (600)
    const [height, setHeightValue] = useState (900)
    const [borderRadius, setborderRadiusValueValue] = useState (15)


    const onWidthInput = (event) => {
        setWidthValue(event.target.value)
    }
    const onHeightInput = (event) => {
        setHeightValue(event.target.value)
    }
    const onborderRadiusInput = (event) => {
        setborderRadiusValueValue(event.target.value)
    }
    const onBannerSizeSubmit = (event) => {
        event.preventDefault()
        const bannerSize = {}
        bannerSize.width = event.target[0].value
        bannerSize.height = event.target[1].value
        bannerSize.borderRadius = event.target[2].value
        bannerSize.ratio = bannerSize.width / bannerSize.height
        props.setBannerSize(bannerSize)

    }

    return (

        <form className ="tools_elem" onSubmit={onBannerSizeSubmit}>
        <h3>Параметры баннера</h3>
            <label htmlFor="widthValue">ширина</label>
            <input type="number" id="widthValue" value = {width} 
                    onChange = {onWidthInput}
                    placeholder = "введите ширину банера" />
            <label htmlFor="heightValue">высота</label>            
            <input type="number" id="heightValue" value = {height} 
                    onChange = {onHeightInput}
                    placeholder = "введите высоту банера" />
            <label htmlFor="borderRadiusValue">скругление</label>
            <input type="number" id="borderRadiusValue" value = {borderRadius} 
                    onChange = {onborderRadiusInput}
                    placeholder = "задайте скругление" />
            <button type="submit">применить</button>
        </form>
    )
}