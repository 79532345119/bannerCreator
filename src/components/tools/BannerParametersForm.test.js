import React from "react"
import {create} from "react-test-renderer"
import {BannerParametersForm} from "./BannerParametersForm"


describe("BannerParametersForm component", ()=> {
    test("BannerParametersForm component renders correctly", ()=> {
        const component = create(<BannerParametersForm/>);
        expect(component).toMatchSnapshot();
    })
})