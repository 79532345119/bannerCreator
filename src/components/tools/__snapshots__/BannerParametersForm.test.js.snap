// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerParametersForm component BannerParametersForm component renders correctly 1`] = `
<form
  className="tools_elem"
  onSubmit={[Function]}
>
  <h3>
    Параметры баннера
  </h3>
  <label
    htmlFor="widthValue"
  >
    ширина
  </label>
  <input
    id="widthValue"
    onChange={[Function]}
    placeholder="введите ширину банера"
    type="number"
    value={600}
  />
  <label
    htmlFor="heightValue"
  >
    высота
  </label>
  <input
    id="heightValue"
    onChange={[Function]}
    placeholder="введите высоту банера"
    type="number"
    value={900}
  />
  <button
    type="submit"
  >
    применить
  </button>
</form>
`;
